graphics.cpp

```diff
@@ -67,3 +46,35 @@ void Graphics::AddSprite(const Sprite& sprite, const SDL_Rect& destination,
   SDL_RenderCopy(renderer_, sprite.GetSpriteTexture(), &source, &destination);
   perf->StopTimer("render_copy");
 }
+
+void Graphics::AddSprite(AnimatedSprite& sprite, Perf* perf,
+                         std::string animation) {
+  if (sprite.GetVisible()) {
+    int frame = sprite.GetFrameIndex();
+    SDL_Rect source = sprite.GetSourceRect(animation, frame);
+    SDL_Rect destination = sprite.GetDestinationRect();
+    Vector2 offset = sprite.GetOffsets(animation);
+    destination.x = (destination.x + offset.x) % (constants::kWindowWidth - 64);
+    destination.y += offset.y;
+    // Draw this sprite on the screen.
+    sprite.SetDestinationRect(destination);
+    perf->StartTimer("render_copy");
+    SDL_RenderCopy(renderer_, sprite.GetSpriteTexture(), &source, &destination);
+    perf->StopTimer("render_copy");
+  }
+}
+
+void Graphics::PlayAnimation(std::string animation, AnimatedSprite& sprite,
```

Comment:
Passing a std::string by value is a no-no since it means you're making a new copy of the string everywhere.

The right way to pass a std::string parameter is either by "const reference" or by using a std::string_view parameter. String_view requires c++17, but it is pretty easy to use:

```
#include <string_view>

// define foo
void foo(std::string_view animation) {}

// use foo
std::string animation = "fart";
foo(animation);
```

Reply:
Although, you really shouldn't be using strings for this. See my other comments about using an enum class.